src/Coverage.TestLogger/CoverageLogger.cs

```diff
@@ -0,0 +1,107 @@
+
+        private const string SetupInteropx64 = @"x64\Microsoft.VisualStudio.Setup.Configuration.Native.dll";
+
+        private static readonly Uri CodeCoverageDataCollectorUri = new Uri(CoverageUri);
+
+        private ManualResetEvent coverageXmlGenerateEvent;
+
+        private CodeCoverageUtility codeCoverageUtility;
+
+        #endregion
+
+        #region ITestLogger
+
+        /// <inheritdoc/>
+        public void Initialize(TestLoggerEvents events, string testResultsDirPath)
+        {
+            if (events == null)
+            {
+                throw new ArgumentNullException(nameof(events));
+            }
+
+            // Register for the events.
+            events.TestRunComplete += this.TestRunCompleteHandler;
+
+            this.codeCoverageUtility = new CodeCoverageUtility();
+        }
+        #endregion
+
+        #region Event Handlers
+
+        /// <summary>
+        /// Called when a test run is completed.
+        /// </summary>
+        /// <param name="sender">
+        /// The sender.
+        /// </param>
+        /// <param name="e">
+        /// Test run complete events arguments.
+        /// </param>
+        internal void TestRunCompleteHandler(object sender, TestRunCompleteEventArgs e)
+        {
+            if (e.AttachmentSets == null)
+            {
+                return;
+            }
+
+            var coverageAttachments = e.AttachmentSets
+                .Where(dataCollectionAttachment => CodeCoverageDataCollectorUri.Equals(dataCollectionAttachment.Uri)).ToArray();
+
+            if (coverageAttachments.Any())
+            {
+                var codeCoverageFiles = coverageAttachments.Select(coverageAttachment => coverageAttachment.Attachments[0].Uri.LocalPath).ToArray();
+                foreach (var codeCoverageFile in codeCoverageFiles)
+                {
+                    var resultFile = Path.Combine(Path.GetDirectoryName(codeCoverageFile), Path.GetFileNameWithoutExtension(codeCoverageFile) + ".xml");
+                    var arguments = "analyze /output:" + '"' + resultFile + '"' + " " + '"' + codeCoverageFile + '"';
+                    try
+                    {
+                        this.codeCoverageUtility.AnalyzeCoverageFile(arguments, this.GetCodeCoverageExePath());
+                    }
+                    catch (Exception ex)
+                    {
+                        ConsoleOutput.Instance.WriteLine(ex.Message, OutputLevel.Information);
+                    }
+
+                    var summary = this.codeCoverageUtility.GetCoverageSummary(resultFile);
+                    ConsoleOutput.Instance.WriteLine(summary, OutputLevel.Information);
+                }
+            }
+        }
+
+        private string GetCodeCoverageExePath()
+        {
+            return @"C:\Users\harjain\.nuget\packages\microsoft.internal.codecoverage\1.0.9\contentFiles\any\any\CodeCoverage";
```

Comment:
We need to find a way to have this directory on every user's machine :sunglasses:

Reply:
Just a thought, shouldn't we convert this to lcov and display the text from there? We could reuse some code from other library.